src/test_utils.jl

```diff
@@ -119,58 +136,75 @@ function StatsBase.fit(
     return FakeModel{E, O}(template.predictor, num_variates)
 end
 
-StatsBase.predict(m::FakeModel, inputs) = m.predictor(m.num_variates, inputs)
+StatsBase.predict(m::FakeModel, inputs::AbstractMatrix) = m.predictor(m.num_variates, inputs)
+StatsBase.predict(m::FakeModel, inputs::AbstractVector{<:Sampleable}) = m.predictor(m.num_variates, inputs)
 
 """
     test_interface(template::Template; inputs=rand(5, 5), outputs=rand(5, 5))
 
 Test that subtypes of [`Template`](@ref) and [`Model`](@ref) implement the expected API.
 Can be used as an initial test to verify the API has been correctly implemented.
 """
-function test_interface(template::Template; kwargs...)
+function test_interface(
+    template::Template; 
+    inputs=rand(5,5), 
+    outputs=_default_outputs(template),
+    distribution_inputs=[MvNormal(5, m) for m in 1:5],
+    kwargs...
+)
     @testset "Models API Interface Test: $(nameof(typeof(template)))" begin
-        test_interface(template, estimate_type(template), output_type(template); kwargs...)
+        predictions = test_common(template, inputs, outputs)
+        test_estimate_type(estimate_type(template), predictions, outputs)
+        test_output_type(output_type(template), predictions, outputs)
+        test_predict_input_type(predict_input_type(template), template, outputs, inputs, distribution_inputs)
     end
 end
 
-function test_interface(
-    template::Template, ::Type{PointEstimate}, ::Type{SingleOutput};
-    inputs=rand(5, 5), outputs=rand(1, 5),
-)
-    predictions = test_common(template, inputs, outputs)
-
-    @test predictions isa NamedDimsArray{(:variates, :observations), <:Real, 2}
-    @test size(predictions) == size(outputs)
-    @test size(predictions, 1) == 1
+function _default_outputs(template)
+    @assert(output_type(template) <: OutputTrait, "Invalid OutputTrait")
+    if output_type(template) == SingleOutput
+        return rand(1, 5)
+    else output_type(template) == MultiOutput
+        return rand(2, 5)
+    end
 end
 
-function test_interface(
-    template::Template, ::Type{PointEstimate}, ::Type{MultiOutput};
-    inputs=rand(5, 5), outputs=rand(2, 5),
-)
-    predictions = test_common(template, inputs, outputs)
+function test_estimate_type(::Type{PointEstimate}, predictions, outputs)
     @test predictions isa NamedDimsArray{(:variates, :observations), <:Real, 2}
     @test size(predictions) == size(outputs)
 end
 
-function test_interface(
-    template::Template, ::Type{DistributionEstimate}, ::Type{SingleOutput};
-    inputs=rand(5, 5), outputs=rand(1, 5),
-)
-    predictions = test_common(template, inputs, outputs)
-    @test predictions isa AbstractVector{<:ContinuousUnivariateDistribution}
+function test_estimate_type(::Type{DistributionEstimate}, predictions, outputs)
+    @test predictions isa AbstractVector{<:ContinuousDistribution}
```

Comment:
Previously this tested where the `Distribution` was `Univariate` or `Multivariate`.  This distinction should now be covered by the tests in `test_output_type`.

Reply:
nice!